src/demo/build.gradle

```diff
@@ -29,7 +29,7 @@ sourceSets.main.java.srcDirs = ['java']
 dependencies {
     implementation 'org.json:json:20201115'
     implementation 'org.springframework.boot:spring-boot-starter-web'
-    compile project(':main')
+    implementation project(':main')
```

Comment:
Good Catch. 
I just found the compile dependency configuration has been removed in the recently released Gradle 7.0.

Reply:
yeah, need to update to get gradle run tests in action